icu4j/main/classes/core/src/com/ibm/icu/impl/locale/LanguageTag.java

```diff
@@ -41,7 +41,7 @@
     // Map contains grandfathered tags and its preferred mappings from
     // http://www.ietf.org/rfc/rfc5646.txt
     private static final Map<AsciiUtil.CaseInsensitiveKey, String[]> GRANDFATHERED =
-        new HashMap<AsciiUtil.CaseInsensitiveKey, String[]>();
```

Comment:
It seems as if some unrelated changes by mistake have been included here as well. The generics changes look wholly unrelated to what the ticket is about.

Reply:
I think I explained this when minimum Java level was bumped to 7. We set up eclipse project to clean up code on save. These generics params are redundant on java 7 or later. For meanwhile, when you edit java files on eclipse, it will happen until.all files are touched. If this is not great, we could make such changes for all ICU4J files. But itwould be better to do such when development activity slows down (either just before release,or after release).

Reply:
My concern here is that if the generics changes are done in the same commit as the language tag changes and we later find out that we'll need to revert the language tag changes then the generics changes would be reverted too, which would be very counter-productive.

Even if Eclipse makes these changes automatically and unconditionally, there is no need or reason to include them in the same commit. It is for example possible to use `git commit -p` to interactively select which of a set of pending changes to include in a commit.

(Including unrelated changes in a commit also breaks down the entire ICU change tracking model of requiring every change to be associated with a ticket, but that's something that I personally feel less concerned about.)

Reply:
I've also added ICU4J fix to #102 without knowing this PR. Should we just go with #102?

Reply:
>My concern here is that if the generics changes are done in the same commit as the language tag changes and we later find out that we'll need to revert the language tag changes then the generics changes would be reverted too, which would be very counter-productive.

I explained this in ICU TC - and did not have any complaint at that time.

>Even if Eclipse makes these changes automatically and unconditionally, there is no need or reason to include them in the same commit.

This means, you should not use eclipse to edit source.

>(Including unrelated changes in a commit also breaks down the entire ICU change tracking model of requiring every change to be associated with a ticket, but that's something that I personally feel less concerned about.)

If this is a concern, please create a ticket to do the transformation for all Java sources.

Reply:
>I've also added ICU4J fix to #102 without knowing this PR. Should we just go with #102?

I'm fine - but you did not compile/test it...

Reply:
The bot did that for me, I guess.